### PR TITLE
refactor(SingleAddSelect): implement typescript types

### PR DESCRIPTION
**Pull request by makafsal**
Contributes to #5117  #5123 #5122 

#### What did you change?
Moved these `.js` files to `.tsx`.
1. `packages/ibm-products/src/components/SingleAddSelect/SingleAddSelect.tsx`
2. `packages/ibm-products/src/components/AddSelect/AddSelect.tsx`
3. `packages/ibm-products/src/components/AddSelect/AddSelectSort.tsx`
4. `packages/ibm-products/src/components/AddSelect/AddSelectBody.tsx`
5. `packages/ibm-products/src/components/Tearsheet/Tearsheet.tsx`
6. `packages/ibm-products/src/components/MultiAddSelect/MultiAddSelect.tsx`

Because all these components are connected, hence I migrated them together to avoid conflicts in future.

#### How did you test and verify your work?
Storybook

### CODE REVIEW

File: packages/ibm-products/src/components/AddSelect/AddSelectBody.tsx [hunk 1]
```diff
@@ -275,11 +351,11 @@ export let AddSelectBody = forwardRef(
...
             {showSort && (
               <AddSelectSort
-                items={itemsToDisplay}
+                // items={itemsToDisplay}
                 setSortAttribute={setSortAttribute}
                 setSortDirection={setSortDirection}
-                sortAttribute={sortAttribute}
-                sortDirection={sortDirection}
+                // sortAttribute={sortAttribute}
+                // sortDirection={sortDirection}
```
Comment (emyarod) [inline packages/ibm-products/src/components/AddSelect/AddSelectBody.tsx:358]: just wondering, are these supposed to stay commented out?

Comment (davidmenendez) [reply packages/ibm-products/src/components/AddSelect/AddSelectBody.tsx:358]: same comment

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks great. one last thing- similarly to the dataspreadsheet refactor, can you move the types into their own file? https://github.com/carbon-design-system/ibm-products/pull/5230/commits/d4c43950e1c761bcfc37b801c662688f8f80c3b3#diff-e0e53665040074874d795a3d1adad8fbb56210c909296066a7e87fcfb25e10b9

**Comment by makafsal** [comment]
@davidmenendez sure, refactored.

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**